maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/utils/ReadFileUtils.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.maven.derive.schema.utils;
+
+import static java.lang.Math.min;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveSchema.mapper;
+
+/**
+ * Utility class for reading files and strings according to input format
+ */
+public class ReadFileUtils {
+
+  private static final Logger logger = LoggerFactory.getLogger(ReadFileUtils.class);
+
+  public static String readFile(String content) throws IOException {
+    byte[] encoded = Files.readAllBytes(Paths.get(content));
+    return new String(encoded, StandardCharsets.UTF_8);
+  }
+
+  /**
+   * Given a file with array of ObjectNodes, reads it and generates List of ObjectNodes.
+   * <p>
+   * Eg, Expected File contents are :
+   * [{"name":"J"}, {"name":"K"}, {"name":"L"}]
+   * </p>
+   *
+   * @param content Content or name of the file to read
+   * @return List of ObjectNodes
+   */
+  public static List<ObjectNode> readArrayOfMessages(String content)
+      throws JsonProcessingException {
+
+    ArrayNode arrayNode = mapper.readValue(content, ArrayNode.class);
+    List<ObjectNode> listOfMessages = new ArrayList<>();
+    for (int i = 0; i < arrayNode.size(); i++) {
+      listOfMessages.add((ObjectNode) arrayNode.get(i));
+    }
```

Comment:
```suggestion
      List<ObjectNode> listOfMessages =  Arrays.asList(mapper.readValue(content, ArrayNode.class));
```

Reply:
```java
Incompatible types. Found: 'java.util.List<com.fasterxml.jackson.databind.node.ArrayNode>', required: 'java.util.List<com.fasterxml.jackson.databind.node.ObjectNode>'
```
``mapper.readValue(content, ArrayNode.class)`` returns ArrayNode but we need read each ObjectNode and add to the list